### PR TITLE
Resolve codacy issues

### PR DESCRIPTION
**Pull request by RadhaGulhane13**
Signed-off-by: Radha Gulhane <radha.gulhane@seagate.com>

# Problem Statement
- Resolve codacy issues

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by shashank-parulekar — COMMENTED** [review]
Can you tell me why is Codacy complaining about these asserts ?

**Comment by RadhaGulhane13** [comment]
> Can you tell me why is Codacy complaining about these asserts?

It was giving a redundancy check warning because the function which allocates node already contains an assert node != NULL and in code, this check was also done by the caller of the function. That is why codacy was giving a redundancy check warning.

File: motr/m0crate/btree_without_tx_seg.c [hunk 1]
```diff
@@ -490,7 +490,6 @@ be_btree_node_alloc(const struct m0_be_btree *btree, struct m0_be_tx *tx)
 	node = (struct m0_be_bnode *)mem_alloc(btree, tx,
 					       sizeof(struct m0_be_bnode),
 					       M0_BITS(M0_BAP_NORMAL));
-	M0_ASSERT(node != NULL);	/* @todo: analyse return code */
```
Comment (nikitadanilov) [inline motr/m0crate/btree_without_tx_seg.c:493]: Please explain, why is this removed. The same for other changes.

**Review by nikitadanilov — CHANGES_REQUESTED** [review]

File: motr/m0crate/btree_without_tx_seg.c (hunk 1 above)
Comment (RadhaGulhane13) [reply motr/m0crate/btree_without_tx_seg.c:493]: These checks (node != NULL) are redundant,  same check is done in mem_alloc, thus codacy was giving following error : 

> Either the condition 'node!=NULL' is redundant or there is a possible null pointer dereference: p_node.

**Comment by RadhaGulhane13** [comment]
Closing the PR as these changes are not required to resolve codacy warning. As per the discussion, these codacy warnings of redundant null pointer check should be ignore rather than removing check from code.
